lib/rhc/commands/ssh.rb

```diff
@@ -42,8 +45,12 @@ def run(_, command)
         $stderr.puts "Connecting to #{rest_app.ssh_string.to_s} ..." unless command.present?
 
         debug "Using user specified SSH: #{options.ssh}" if options.ssh
-
-        command_line = [RHC::Helpers.split_path(ssh), ('-vv' if debug?), rest_app.ssh_string.to_s, command].flatten.compact
+        debug "Using user specified SSH Private Key: #{options.key}" if options.key
+        if options.key
+          command_line = [RHC::Helpers.split_path(ssh), ('-vv' if debug?), rest_app.ssh_string.to_s, "-i", options.key, command].flatten.compact
+        else
+          command_line = [RHC::Helpers.split_path(ssh), ('-vv' if debug?), rest_app.ssh_string.to_s, command].flatten.compact
+        end
```

Comment:
This could be simplified by using something like `(["-i", options.key] if options.key?)`.  The `.flatten` will flatten it out if `options.key?` is true, and the `.compact` will delete it if `options.key?` is false.